From dcf1099d89de0ff0e4488f29b563607f3e4f244f Mon Sep 17 00:00:00 2001
From: "Anna (navi) Figueiredo Gomes" <navi@vlhl.dev>
Date: Fri, 25 Apr 2025 18:44:53 +0200
Subject: [PATCH] init.d/meson.build: install agetty as a relative symlink

it's the standard for symlinks to exist as relative pointers. absolute
symlinks at install time also triggers an issue with meson[1]

1: https://github.com/mesonbuild/meson/issues/12253
Fixes: https://github.com/OpenRC/openrc/issues/847
---
 init.d/meson.build | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/init.d/meson.build b/init.d/meson.build
index 4316ebd6..a0e78a66 100644
--- a/init.d/meson.build
+++ b/init.d/meson.build
@@ -105,8 +105,7 @@ if get_option('sysvinit') == true and os == 'Linux'
   default_runlevel_dir = get_option('sysconfdir') / 'runlevels' / 'default'
   foreach x : get_option('agetty')
     link_name = 'agetty.' + x
-    install_symlink(link_name, install_dir: init_d_dir,
-      pointing_to: init_d_dir / 'agetty')
+    install_symlink(link_name, install_dir: init_d_dir, pointing_to: 'agetty')
     install_symlink(link_name, install_dir: default_runlevel_dir,
       pointing_to: init_d_dir / link_name)
   endforeach
